app/controllers/api/v1/movies_controller.rb

```diff
@@ -9,6 +9,13 @@ def index
     render json: @movies
   end
 
+  def search_by_imdb_id
+    movie = Rails.cache.fetch("movies/#{params[:id]}/search_by_imdb_id", expires_in: 12.hours) do

```

Comment:
Just for your information: we can setup Rails cache to use memory, redis, or any other custom adapter to store our keys.